client/client-multi/client-c/src/kaa/kaa_notification_manager.h

```diff
@@ -0,0 +1,185 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KAA_KAA_NOTIFICATION_H_
+#define KAA_KAA_NOTIFICATION_H_
+#include <stddef.h>
+#include <stdio.h>
+#include <stdint.h>
+#include "stdbool.h"
+#include "kaa_status.h"
+#include "kaa_error.h"
+#include "kaa_platform_common.h"
+#include "utilities/kaa_mem.h"
+#include "kaa_channel_manager.h"
+#include "kaa_common.h"
+#include "utilities/kaa_log.h"
+#include "kaa_platform_utils.h"
+#include "platform/ext_notification_receiver.h"
+
+
+/**
+* @brief Calculates the topic listener id to manage topic listeners.
+*
+* @param[in]   listener     Pointer to the listener, which id should be calculated.
+* @param[out]  listener_id  Pointer to the variable, which will be initialized with calculated id.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_calculate_topic_listener_id(kaa_topic_listener_t *listener, uint32_t *listener_id);
+
+/**
+* @brief Calculates the notification listener id to manage topic listeners.
+*
+* @param[in]   listener     Pointer to the listener, which id should be calculated.
+* @param[out]  listener_id  Pointer to the variable, which will be initialized with calculated id.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_calculate_notification_listener_id(kaa_notification_listener_t *listener, uint32_t *listener_id);
+
+/**
+* @brief Adds a mandatory notification listener.
+*
+* @param[in]  self         Pointer to the notification manager instance.
+* @param[in]  listener     Pointer to the listener, which callback will be called as soon as a notification received.
+* @param[out] listener_id  Pointer to the variable, which will be initialized with calculated id.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_add_notification_listener(kaa_notification_manager_t *self, kaa_notification_listener_t *listener, uint32_t *listener_id);
+
+/**
+* @brief Adds an optional notification listener.
+*
+* @param[in]  self        Pointer to the notification manager instance.
+* @param[in]  listener    Pointer to a listener, which callback will be called as soon as a notification received.
+* @param[in]  topic_id    Pointer to an id of topic, listener wants to be notified about.
+* @param[out] listener_id Pointer to an variable, which will be initialized with calculated id. If NULL listener_id won't be initialized
+*
+* @return Error code.
+*/
+kaa_error_t kaa_add_optional_notification_listener(kaa_notification_manager_t *self, kaa_notification_listener_t *listener
+                                                 , uint64_t *topic_id, uint32_t *listener_id);
+
+/**
+* @brief Removes a mandatory notification listener
+*
+* @param[in]  self         Pointer to the notification manager instance.
+* @param[in]  listener_id  Pointer to the listener id, which will be used to find listener that should be removed from notification listeners' list. If NULL listener_id won't be initialized
```

Comment:
The pointer to the listener id which is used to find the listener that should be removed from the notification listeners list. If NULL, the listener_id is not initialized.